solana/rpc/websocket.py

```diff
@@ -0,0 +1,207 @@
+"""API WebSocket Client to interact with the Solana JSON RPC Endpoint."""
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional, Union
+from websockets import WebSocketClientProtocol  # type: ignore
+
+from solana.publickey import PublicKey
+
+from .commitment import Commitment, Max
+from .providers.ws import WSProvider
+from .types import RPCMethod, RPCResponse
+
+
+class WebSocketClient:
+    """Websocket client class."""
+
+    _comm_key = "commitment"
+    _encoding_key = "encoding"
+    _data_slice_key = "dataSlice"
+    _filters_key = "filters"
+
+    def __init__(self, endpoint: str):
+        """Init API client for websocket."""
+        self._provider = WSProvider(endpoint)
+        self._conn = self._provider.conn.ws
+
+    def connection(self) -> Optional[WebSocketClientProtocol]:
```

Comment:
```suggestion
    def connect(self) -> WebSocketClientProtocol:
```

To be consistent with https://websockets.readthedocs.io/en/stable/api.html#module-websockets.client.